os/fs/vfs/fs_pwrite.c

```diff
@@ -111,7 +111,10 @@ ssize_t file_pwrite(FAR struct file *filep, FAR const void *buf, size_t nbytes,
 	/* Then perform the write operation */
 
 	ret = file_write(filep, buf, nbytes);
-	errcode = get_errno();
+	if (ret < 0) {
```

Comment:
Please also take care below code snippet in aio_wrint.c at line 182 ~ 189.
```c
	/* Check the result of the write */

	if (nwritten < 0) {
		int errcode = get_errno();
		fdbg("ERROR: write/pwrite failed: %d\n", errcode);
		DEBUGASSERT(errcode > 0);
		aiocbp->aio_result = -errcode;
	} ...
```

Reply:
In addition, on nuttx, file_pwrite() does not set errno, but return negated error code as file_write().